resources/src/main/java/org/robolectric/res/android/ResourceString.java

```diff
@@ -82,22 +82,26 @@ public static String decodeString(ByteBuffer buffer, int offset, Type type) {
     } else {
       length = characterCount * 2;
     }
-    ByteBuffer stringBuffer = ByteBuffer.wrap(buffer.array(), offset, length);
-    // Use normal UTF-8 and UTF-16 decoder to decode string
-    try {
-      return type.decoder().decode(stringBuffer).toString();
-    } catch (CharacterCodingException e) {
-      if (type == Type.UTF16) {
+    if (buffer.hasArray()) {
```

Comment:
I think it's safe to suppress this warning -- we create and control all of the ByteBuffers here.

Reply:
ah okay, Yeah we can suppress it, then the little trade-off is that now it has to keep in mind a thingy.

Reply:
For now lme suppress it